twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt

```diff
@@ -21,7 +30,75 @@ class TwingraphServiceImpl : TwingraphApiService {
     TODO("Not yet implemented")
   }
 
-  override fun delete() {
-    TODO("Not yet implemented")
+  override fun delete(graphId: String) {
+    val matchingKeys = mutableSetOf<String>()
+    var nextCursor = SCAN_POINTER_START
+    do {
+      var scanResult = jedis.scan(nextCursor, ScanParams().match("$graphId*"))
+      nextCursor = scanResult.cursor
+      matchingKeys.addAll(scanResult.result)
+    } while (!nextCursor.equals(SCAN_POINTER_START))
+
+    @Suppress("SpreadOperator") val count = jedis.del(*matchingKeys.toTypedArray())
+    logger.debug("$count keys are removed from Twingraph with prefix $graphId")
+  }
+
+  override fun query(organizationId: String, twinGraphQuery: TwinGraphQuery): String {
+
+    if (twinGraphQuery.version == null) {
+      twinGraphQuery.version = jedis.hget("${twinGraphQuery.graphId}MetaData", "lastVersion")
+    }
```

Comment:
Handle case if lastVersion is not found or null. 
You can throw an exception (e.g. IllegalStateException)